syntax = "proto3";
package protos;
import "spaghetr/protos/wrappers.proto";
import "spaghetr/protos/process.proto";


service Subprocess {
    rpc Run (ArgsRequest) returns (StdReply) {}
    rpc Popen (ArgsRequest) returns (StdReply) {}

}


// Basic response from a process
message StdReply {
    // exit code
    sint32 returncode = 1;

    // PID of process
    sint32 pid = 2;

    // output of stdout
    bytes stdout = 3;

    // output of stderr
    bytes stderr = 4;
}
